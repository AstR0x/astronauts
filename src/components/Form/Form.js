import React, { useState } from 'react';

import { MILLISECONDS_IN_SECOND } from '../../constants';

import styles from './Form.module.css';

const initialAstronaut = {
  name: '',
  date: '',
  days: '',
  mission: '',
};

export const Form = ({ onAdd }) => {
  const [astronaut, setAstronaut] = useState(initialAstronaut);
  const [isMultiple, setIsMultiple] = useState(false);

  const onChangeIsMultiple = event => setIsMultiple(event.target.checked);
  const onChangeAstronaut = event => setAstronaut({ ...astronaut, [event.target.name]: event.target.value });

  const isValid = astronaut.name.trim()
    && astronaut.date
    && astronaut.days
    && astronaut.mission.trim();

  const onSubmitForm = event => {
    event.preventDefault();
    const { name, date, days, mission } = astronaut;
    const timestamp = new Date(date).getTime() / MILLISECONDS_IN_SECOND;
    const id = Date.now();

    onAdd({
      id,
      name,
      date: timestamp,
      days: parseInt(days, 10),
      mission,
      isMultiple,
    });

    setAstronaut(initialAstronaut);
    setIsMultiple(false);
  };

  const inputs = [
    { value: astronaut.name, name: 'name', type: 'text', placeholder: 'Имя' },
    { value: astronaut.date, name: 'date', type: 'date', placeholder: 'Дата' },
    { value: astronaut.days, name: 'days', type: 'number', placeholder: 'Дней в космосе' },
    { value: astronaut.mission, name: 'mission', type: 'text', placeholder: 'Название миссии' }];

  return (
    <form onSubmit={onSubmitForm} className={styles.form}>
      {inputs.map(({ value, type, placeholder, name }) => (
        <input
          type={type}
          value={value}
          name={name}
          onChange={onChangeAstronaut}
          className={styles.input}
          placeholder={placeholder}
          key={placeholder}
          required
        />
      ))}
      <div className={styles.checkboxContainer}>
        <label className={styles.checkboxLabel}>
          Повторные полёты
        </label>
        <input
          checked={isMultiple}
          type="checkbox"
          onChange={onChangeIsMultiple}
          className={styles.checkbox}
        />
      </div>
      <button
        disabled={!isValid}
        className={styles.button}
        type="submit"
      >
        Добавить
      </button>
    </form>
  );
};
