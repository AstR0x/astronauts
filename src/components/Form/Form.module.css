.form {
  display: flex;
  justify-content: space-between;
  margin-top: 30px;
}

.input {
  padding-left: 10px;
  height: 35px;
  width: 15%;
  border: 1px solid #ccc;
  border-radius: 4px;
  outline: none;
}

.checkboxContainer {
  display: flex;
  align-items: center;
}

.checkboxLabel {
  margin-right: 10px;
  font-size: 14px;
}

.checkbox {
  height: 20px;
  width: 20px;
}

.button {
  background-color: #fff;
  border: 1px solid #ccc;
  border-radius: 2px;
}