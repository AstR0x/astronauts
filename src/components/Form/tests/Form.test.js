import React from 'react';
import renderer from 'react-test-renderer';

import { Form } from 'components/Form';

describe('component Form should', () => {
  const component = renderer.create(<Form />);
  const tree = component.toJSON();

  it('correct render', () => {
    expect(tree).toMatchSnapshot();
  });
});
