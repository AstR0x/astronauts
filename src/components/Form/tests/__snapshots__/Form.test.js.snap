// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Form should correct render 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="input"
    onChange={[Function]}
    placeholder="Имя"
    required={true}
    type="text"
    value=""
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="Дата"
    required={true}
    type="date"
    value=""
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="Дней в космосе"
    required={true}
    type="number"
    value=""
  />
  <input
    className="input"
    onChange={[Function]}
    placeholder="Название миссии"
    required={true}
    type="text"
    value=""
  />
  <div
    className="checkboxContainer"
  >
    <label
      className="checkboxLabel"
    >
      Повторные полёты
    </label>
    <input
      checked={false}
      className="checkbox"
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
  </div>
  <button
    className="button"
    disabled={true}
    type="submit"
  >
    Добавить
  </button>
</form>
`;
